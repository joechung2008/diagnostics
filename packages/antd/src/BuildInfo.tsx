import { Table } from "antd";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const data = [
    {
      key: 0,
      name: "Build Version",
      value: buildVersion,
    },
  ];

  const columns = [
    {
      title: "Name",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Value",
      dataIndex: "value",
      key: "value",
    },
  ];

  return (
    <Table
      aria-label="Build Info"
      columns={columns}
      dataSource={data}
      pagination={false}
    />
  );
};

export default BuildInfo;
