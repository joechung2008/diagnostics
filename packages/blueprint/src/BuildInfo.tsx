import { HTMLTable } from "@blueprintjs/core";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  return (
    <HTMLTable aria-label="Build Info">
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item) => (
          <tr key={item.name}>
            <td>{item.name}</td>
            <td>{item.value}</td>
          </tr>
        ))}
      </tbody>
    </HTMLTable>
  );
};

export default BuildInfo;
