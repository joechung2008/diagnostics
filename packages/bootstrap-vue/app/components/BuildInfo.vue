<template>
  <div>
    <BTable :items="rows" :fields="fields" small />
  </div>
</template>

<script setup lang="ts">
import type { BuildInfoProps } from "@/types/BuildInfoProps";

const props = defineProps<BuildInfoProps>();

const fields = [
  { key: "name", label: "Name" },
  { key: "value", label: "Value" }
];

const rows = computed(() => [{ name: "Build Version", value: props.buildVersion }]);
</script>
