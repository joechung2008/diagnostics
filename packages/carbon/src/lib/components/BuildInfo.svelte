<script lang="ts">
	import { DataTable } from 'carbon-components-svelte';

	const { buildVersion }: BuildInfoProps = $props();

	const headers = [
		{ key: 'name', value: 'Name', empty: false },
		{ key: 'value', value: 'Value', empty: false }
	];

	const rows = $derived([{ id: 'build-version', name: 'Build Version', value: buildVersion }]);
</script>

<DataTable aria-label="Build Information" {headers} {rows} />
