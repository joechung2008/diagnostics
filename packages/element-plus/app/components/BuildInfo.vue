<template>
    <el-table :data="rows" style="width: 100%" aria-label="Build Info">
        <el-table-column prop="name" label="Name" show-overflow-tooltip />
        <el-table-column prop="value" label="Value" show-overflow-tooltip />
    </el-table>
</template>

<script setup lang="ts">
import type { BuildInfoProps } from "@/types"

const props = defineProps<BuildInfoProps>()

const rows = computed(() => [
    {
        name: "Build Version",
        value: props.buildVersion
    }
])
</script>
