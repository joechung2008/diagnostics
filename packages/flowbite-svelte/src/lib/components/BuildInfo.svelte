<script lang="ts">
  import {
    Table,
    TableBody,
    TableBodyCell,
    TableBodyRow,
    TableHead,
    TableHeadCell
  } from "flowbite-svelte";

  let { buildVersion }: BuildInfoProps = $props();

  const items = $derived.by(() => [{ name: "Build Version", value: buildVersion }]);
</script>

<Table aria-label="Build Info">
  <TableHead>
    <TableHeadCell>Name</TableHeadCell>
    <TableHeadCell>Value</TableHeadCell>
  </TableHead>
  <TableBody>
    {#each items as item (item.name)}
      <TableBodyRow>
        <TableBodyCell>{item.name}</TableBodyCell>
        <TableBodyCell>{item.value}</TableBodyCell>
      </TableBodyRow>
    {/each}
  </TableBody>
</Table>
