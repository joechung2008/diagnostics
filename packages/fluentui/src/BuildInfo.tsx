import {
  Table,
  TableHeader,
  TableRow,
  TableHeaderCell,
  TableBody,
  TableCell,
} from "@fluentui/react-components";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  return (
    <Table aria-label="Build Info">
      <TableHeader>
        <TableRow>
          <TableHeaderCell>Name</TableHeaderCell>
          <TableHeaderCell>Value</TableHeaderCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default BuildInfo;
