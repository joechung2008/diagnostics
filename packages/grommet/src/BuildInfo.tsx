import { Table, TableBody, TableCell, TableHeader, TableRow } from "grommet";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  return (
    <Table aria-label="Build Info">
      <TableHeader>
        <TableRow>
          <TableCell scope="col">Name</TableCell>
          <TableCell scope="col">Value</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default BuildInfo;
