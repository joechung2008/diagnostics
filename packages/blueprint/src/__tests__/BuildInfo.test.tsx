import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import BuildInfo from "../BuildInfo";

describe("BuildInfo", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<BuildInfo buildVersion="1.0.0" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
