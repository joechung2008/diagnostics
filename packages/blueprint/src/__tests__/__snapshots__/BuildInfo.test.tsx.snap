// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildInfo > matches snapshot 1`] = `
<DocumentFragment>
  <table
    aria-label="Build Info"
    class="bp6-html-table"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Build Version
        </td>
        <td>
          1.0.0
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
